### PR TITLE
make Amber easyblock aware of FlexiBLAS

### PR DESCRIPTION
**Pull request by boegel**
(created using `eb --new-pr`)

### CODE REVIEW

File: easybuild/easyblocks/a/amber.py [hunk 1]
```diff
@@ -141,6 +141,17 @@ def configure_step(self):
         if self.toolchain.options.get('openmp', None):
             self.cfg.update('configopts', '-DOPENMP=TRUE')
 
+        # note: for Amber 20, a patch is required to fix the CMake scripts so they're aware of FlexiBLAS:
+        # - cmake/patched-cmake-modules/FindBLASFixed.cmake
+        # - cmake/patched-cmake-modules/FindLAPACKFixed.cmake
```
Comment (akesandgren) [inline easybuild/easyblocks/a/amber.py:146]: Since this code is traversed for Amber 19 too, don't we need a version check here?

Comment (boegel) [reply easybuild/easyblocks/a/amber.py:146]: It should be harmless as is: if you're using a toolchain that includes FlexiBLAS, you'll use `-DBLA_VENDOR=FlexiBLAS`, which will cause a clear configuration error if the patches are not included (they most likely also work with Amber 19, I didn't check).
The patch is available in https://github.com/fwautele/easybuild-easyconfigs/pull/1, so it should become part of https://github.com/easybuilders/easybuild-easyconfigs/pull/14386 soon (the patch for `Amber` 20 is exactly the same as for AmberTools 21).

If you're using a toolchain that doesn't include FlexiBLAS but does include OpenBLAS, you'll use `-DBLA_VENDOR=OpenBLAS`, which is what the CMake script does automatically already (we're just making that explicit now and taking control).

For other toolchains, we don't specify `-DBLA_VENDOR` and we leave it up to the CMake scripts to figure out which BLAS/LAPACK to use, which is also what happens without these changes.

Long story short: none of the existing easyconfigs should be "hurt" by the changes being made here...

File: easybuild/easyblocks/a/amber.py [hunk 2]
```diff
@@ -213,10 +224,11 @@ def configuremake_install_step(self):
...
-        openblasroot = get_software_root('OpenBLAS')
+        flexiblas_root = get_software_root('FlexiBLAS')
+        openblas_root = get_software_root('OpenBLAS')
         if mklroot:
             env.setvar('MKL_HOME', os.getenv('MKLROOT'))
-        elif openblasroot:
+        elif flexiblas_root or openblas_root:
             lapack = os.getenv('LIBLAPACK')
             if lapack is None:
                 raise EasyBuildError("LIBLAPACK (from OpenBLAS) not found in environment.")
```
Comment (akesandgren) [inline easybuild/easyblocks/a/amber.py:234]: Should we change this error to mention the correct BLAS? i.e. FlexiBLAS or OpenBLAS?

Comment (akesandgren) [reply easybuild/easyblocks/a/amber.py:234]: ping @boegel

Comment (boegel) [reply easybuild/easyblocks/a/amber.py]: @akesandgren fixed in effecb7c1

**Review by akesandgren — APPROVED** [review]
LGTM

**Comment by akesandgren** [comment]
Going in, thanks @boegel!
